internal/captain/command.go

```diff
@@ -218,6 +240,38 @@ func (c *Command) AddLegacyChildren(children ...cobraCommander) {
 	}
 }
 
+func (c *Command) Find(args []string) *Command {
+	var innerFind func(*Command, []string) (*Command, []string)
+
+	innerFind = func(innerCmd *Command, innerArgs []string) (*Command, []string) {
+		if len(innerArgs) == 0 {
+			return innerCmd, innerArgs
+		}
+
+		cmd := innerCmd.findNext(innerArgs[0])
+		if cmd != nil {
+			return innerFind(cmd, innerArgs[1:])
+		}
+		return innerCmd, innerArgs
+	}
+
+	if len(args) < 2 {
+		return c
+	}
+
+	found, _ := innerFind(c, args[1:])
+	return found
+}
+
+func (c *Command) findNext(next string) *Command {
+	for _, cmd := range c.commands {
+		if cmd.cobra.Use == next {
+			return cmd
+		}
+	}
+	return nil
+}
```

Comment:
Why not use `cobra.Find`? This needs a better solution regardless, we're doing too much here.

Reply:
For the purpose of the skip checks we are setting properties on `captain.Command`. `cobra.Find` will return us a `cobra.Command` and we do not have a way to convert a cobra command to one of ours.

Reply:
I think a less involved and more scalable solution would be to keep a mapping of cobra <-> our commands and use that in tandem with `cobra.Find`

Reply:
Makes sense, it's wrapped now.